src/frontegg-mock/src/lib.rs

```diff
@@ -376,6 +384,130 @@ async fn handle_post_user_api_token<'a>(
     Ok(Json(new_token))
 }
 
+// https://docs.frontegg.com/reference/userscontrollerv1_getuserbyid
+async fn handle_get_user(
+    State(context): State<Arc<Context>>,
+    Path(user_id): Path<Uuid>,
+) -> impl IntoResponse {
+    let users = context.users.lock().unwrap();
+    let role_mapping = get_role_mapping();
+
+    match users.iter().find(|(_, user)| user.id == Some(user_id)) {
+        Some((_, user)) => {
+            // Convert the stored role names in UserConfig to UserRole structs.
+            let roles: Vec<UserRole> = user.roles.iter().map(|role_name| {
+                role_mapping.get(role_name).cloned().unwrap_or_else(|| UserRole {
+                    id: "unknown".to_string(),
+                    name: role_name.to_string(),
+                })
+            }).collect();
+
+            // Construct and return the UserResponse.
+            let user_response = UserResponse {
+                id: user.id.unwrap_or_default(),
+                email: user.email.clone(),
+                verified: user.verified.unwrap_or(false),
+                metadata: user.metadata.clone().unwrap_or_default(),
+                provider: user.auth_provider.clone().unwrap_or_default(),
+                roles,
+            };
+
+            Json(user_response).into_response()
+        },
```

Comment:
There is some additional logic here mainly because the current setup of the `UserConfig` stores the roles as `Vec<String>`, however according to the API docs, the `roles` is an array of objects.

Not 100% sure about why the current mock implementation uses a `Vec<String>`, so I've decided to leave it untouched and add this additional logic here to construct the response expected by the Terraform provider.

<img width="683" alt="image" src="https://github.com/MaterializeInc/materialize/assets/21223421/eb6bbd77-32e9-49dd-aab3-510e909bf0bf">